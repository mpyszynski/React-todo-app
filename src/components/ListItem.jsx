import React from 'react';
import PropTypes from 'prop-types';

export default function ListItem(props) {
  const { title, description, isCompleted, id } = props.todo;
  const liClass = isCompleted
    ? 'todo__item todo__item--completed'
    : 'todo__item';
  return (
    <li onClick={() => props.toggleCompletion(id)} className={liClass}>
      <section className='todo__item--section'>
        <h1>{title}</h1>
        <p>{description}</p>
      </section>
      {isCompleted && (
        <div className='todo__item--div'>
          <button
            onClick={e => props.deleteItem(e, id)}
            className='todo__item--button'
          >
            Remove
          </button>
        </div>
      )}
    </li>
  );
}

ListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  deleteItem: PropTypes.func.isRequired,
  toggleCompletion: PropTypes.func.isRequired
};
