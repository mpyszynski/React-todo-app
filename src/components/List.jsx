import React from 'react';
import ListItem from './ListItem';
import './style/list.css';
import PropTypes from 'prop-types';

export default function List(props) {
  const listRender = props.todoList.map(listItem => (
    <ListItem
      key={listItem.id}
      todo={listItem}
      toggleCompletion={props.toggleCompletion}
      deleteItem={props.deleteItem}
    />
  ));
  return <ul className='todo__main'>{listRender}</ul>;
}

List.propTypes = {
  toggleCompletion: PropTypes.func.isRequired,
  deleteItem: PropTypes.func.isRequired,
  todoList: PropTypes.array.isRequired,
};
