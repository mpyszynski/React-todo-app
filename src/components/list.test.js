import React from 'react';
import { shallow } from 'enzyme';
import ListItem from './ListItem';
import List from './List';

let mockArray = [];
const mockFunction = jest.fn();
beforeEach(() => {
  mockArray = [
    {
      title: 'Exercise',
      description: 'Go to gym and do some lifting',
      id: 3,
      isCompleted: false,
    },
    {
      title: 'Buy chocolate',
      description: 'Go to Kiwi and buy some chocolate',
      id: 4,
      isCompleted: true,
    },
    {
      title: 'Buy cola',
      description: 'Go to Kiwi and buy some cola',
      id: 2,
      isCompleted: true,
    },
    {
      title: 'Buy milk',
      description: 'Go to Kiwi and buy some milk',
      id: 1,
      isCompleted: false,
    },
  ];
});

describe('<ListItem />', () => {
  test('should return item with correct information', () => {
    const entry = shallow(
      <ListItem
        todo={mockArray[0]}
        toggleCompletion={mockFunction}
        deleteItem={mockFunction}
      />,
    );
    expect(entry.find('.todo__item--section h1').text()).toEqual('Exercise');
    expect(entry.find('.todo__item--section p').text()).toEqual(
      'Go to gym and do some lifting',
    );
  });
  test('should return item list without button', () => {
    const entry = shallow(
      <ListItem
        todo={mockArray[0]}
        toggleCompletion={mockFunction}
        deleteItem={mockFunction}
      />,
    );
    const buttonDiv = entry.find('.todo__item--div');
    expect(entry).toMatchSnapshot();
    expect(entry.hasClass('todo__item')).toBeTruthy();
    expect(entry.hasClass('todo__item--completed')).toBeFalsy();
    expect(buttonDiv.length).toEqual(0);
  });
  test('should return item list with button', () => {
    const entry = shallow(
      <ListItem
        todo={mockArray[1]}
        toggleCompletion={mockFunction}
        deleteItem={mockFunction}
      />,
    );
    const buttonDiv = entry.find('.todo__item--div');
    expect(entry).toMatchSnapshot();
    expect(entry.hasClass('todo__item--completed')).toBeTruthy();
    expect(buttonDiv.length).toEqual(1);
  });
});

describe('<List />', () => {
  test('should return list with 4 elements', () => {
    const wrapper = shallow(
      <List
        todoList={mockArray}
        toggleCompletion={mockFunction}
        deleteItem={mockFunction}
      />,
    );
    expect(wrapper.find('.todo__main').children().length).toEqual(4);
    expect(wrapper).toMatchSnapshot();
  });
});
