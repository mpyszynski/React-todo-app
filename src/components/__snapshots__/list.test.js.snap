// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should return list with 4 elements 1`] = `ShallowWrapper {}`;

exports[`<ListItem /> should return item list with button 1`] = `ShallowWrapper {}`;

exports[`<ListItem /> should return item list without button 1`] = `ShallowWrapper {}`;
